qemu/tests/snp_multi_vm.py

```diff
@@ -0,0 +1,74 @@
+import os
+
+from virttest import data_dir as virttest_data_dir
+from virttest import env_process, error_context
+from virttest.utils_misc import get_mem_info, normalize_data_size, verify_dmesg
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Qemu snp basic test on Milan and above host:
+    1. Check host snp capability
+    2. Boot snp VM
+    3. Verify snp enabled in guest
+    4. Check snp qmp cmd and policy
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    error_context.context("Start sev-snp test", test.log.info)
+    timeout = params.get_numeric("login_timeout", 240)
+
+    snp_module_path = params["snp_module_path"]
+    if os.path.exists(snp_module_path):
+        with open(snp_module_path) as f:
+            output = f.read().strip()
+        if output not in params.objects("module_status"):
+            test.cancel("Host sev-snp support check fail.")
+    else:
+        test.cancel("Host sev-snp support check fail.")
+    # Define vm memory size for multi vcpus scenario
+    if params.get_numeric("smp") > 1:
+        MemFree = float(
+            normalize_data_size("%s KB" % get_mem_info(attr="MemFree"), "M")
+        )
+        vm_num = len(params.get("vms").split())
+        params["mem"] = MemFree // (2 * vm_num)
+
+    params["start_vm"] = "yes"
+    vms = params.objects("vms")
+    for vm_name in vms:
+        params["mem"] = 4096
+        env_process.preprocess_vm(test, params, env, vm_name)
```

Comment:
remove this line

Reply:
if removing preprocess, the memory defined in the tests won't take effects.
e.g. the mem size will be the same with the profile defined without preprocess.
    vms = params.objects("vms")
    for vm_name in vms:
        vm = env.get_vm(vm_name)
        vm.create()
        vm.verify_alive()

Reply:
@zhencliu updated the patch could you please help to review it?
Checked the guests' memory and cpus are the correct setting.

Reply:
we can change the vm's params(i.e. vm.params["mem"]) instead of modifying the global params, then we don't need to preprocess vm again, but anyway, your code is also OK for me